code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -576,10 +576,17 @@
 	glass_name = "soy latte"
 	glass_desc = "A nice and refreshing beverage while you're reading."
 
+/datum/reagent/consumable/soy_latte/on_mob_add(mob/living/M)
+	M.add_trait(TRAIT_SLEEPIMMUNE)
```

Comment:
but what if you're an elf

Reply:
Is there a joke I'm missing here? What about elves and cafe latte?

Reply:
space elves next pr

Reply:
only if eating an elf corpse gives you sleep resistance, a la nethack

Reply:
you need to specify the source of the trait to add one, and it needs to be unique so it doesn't get overridden by another source